pkg/util/grunning/enabled_test.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// See grunning.Supported() for an explanation behind this build tag.
+//
+//go:build !((darwin && arm64) || freebsd || !bazel)
+// +build !darwin !arm64
+// +build !freebsd
+// +build bazel
+
+package grunning_test
+
+import (
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
+	"github.com/cockroachdb/cockroach/pkg/util/grunning"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/stretchr/testify/require"
+)
+
+// This file tests various properties we should expect over the running time
+// value. It does not make assertions given the CI environments we run these
+// under (CPU-starved, lot of OS thread pre-emption, dissimilar to healthy CRDB
+// deployments). This is also why they're skipped under stress. Still, these
+// tests are useful to understand the properties we expect running time to have.
+
+func TestEnabled(t *testing.T) {
+	require.True(t, grunning.Supported())
+	j := 1
+	for i := 0; i < 5000000000; i++ {
+		j = j - i + 42
```

Comment:
Hmmm. Not sure about this test. IDK about the Go compiler but a lot of compilers can pre-compute constants like this. If the Go compiler can't do it yet due to being too dumb, I'd say it's only a matter of time. I would say delete this test and just add `require.True(t, grunning.Supported())` to another one of the tests in this file.

Reply:
Removed, though the busy loops below too will get defeated by future Go compiler smartness. Probably we'll come up with more useful tests by then.